File: LLM-Translations/GPT-4/codenet/Java/Python/s302760734.py
import sys

def main():
    s = list(input().strip())
    t = list(input().strip())
    check = False
    for i in range(len(s) - len(t) + 1):
        check2 = True
        for j in range(len(t)):
            if s[i + j] != '?' and s[i + j] != t[j]:
                check2 = False
        if check2:
            check = True
            for j in range(len(t)):
                s[i + j] = t[j]
        if check:
            break
    if check:
        for i in range(len(s)):
            if s[i] == '?':
                sys.stdout.write("a")
            else:
                sys.stdout.write(s[i])
        sys.stdout.write("\n")
    else:
        print("UNRESTORABLE")

if __name__ == "__main__":
    main()

File: LLM-Translations/GPT-4/codenet/Java/C/s302760734.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

int main() {
    char s[51], t[51];
    scanf("%s %s", s, t);

    int s_length = strlen(s);
    int t_length = strlen(t);
    bool check = false;

    for (int i = s_length - t_length; i >= 0; i--) {
        bool check2 = true;
        for (int j = 0; j < t_length; j++) {
            if (s[i + j] != '?' && s[i + j] != t[j]) {
                check2 = false;
            }
        }
        if (check2) {
            check = true;
            for (int j = 0; j < t_length; j++) {
                s[i + j] = t[j];
            }
        }
        if (check) {
            break;
        }
    }

    if (check) {
        for (int i = 0; i < s_length; i++) {
            if (s[i] == '?') {
                printf("a");
            } else {
                printf("%c", s[i]);
            }
        }
        printf("\n");
    } else {
        printf("UNRESTORABLE\n");
    }

    return 0;
}

// 
